src/coffee/helpers/task-queue.coffee

```diff
@@ -49,13 +50,19 @@ class TaskQueue
     @_activeCount += 1
 
     task.fn()
-    .then (res) ->
-      task.resolve res
-    .catch (error) ->
-      task.reject error
+    .then (res) =>
+      # postpone resolving the task
+      @_completedTasks.push(task.resolve.bind(this, res))
+    .catch (error) =>
+      # postpone rejecting the task
+      @_completedTasks.push(task.reject.bind(this, error))
```

Comment:
but just thinking  this again, I think it's better we reject if any error should occur, doesn't make sense to reject and resolve in the same execution, it will be more confusing for the user and for error handling so I go with the way @butenkor propose...